src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -138,6 +138,172 @@ Convolution<
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::Convolution(
+    Convolution&& layer) : 
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    batchSize(std::move(layer.batchSize)),
+    kernelWidth(std::move(layer.kernelWidth)),
+    kernelHeight(std::move(layer.kernelHeight)),
+    strideWidth(std::move(layer.strideWidth)),
+    strideHeight(std::move(layer.strideHeight)),
+    padWLeft(std::move(layer.padWLeft)),
+    padWRight(std::move(layer.padWRight)),
+    padHBottom(std::move(layer.padHBottom)),
+    padHTop(std::move(layer.padHTop)),
+    weights(std::move(layer.weights)),
+    weight(std::move(layer.weight)),
+    bias(std::move(layer.bias)),
+    inputWidth(std::move(layer.inputWidth)),
+    inputHeight(std::move(layer.inputHeight)),
+    outputWidth(std::move(layer.outputWidth)),
+    outputHeight(std::move(layer.outputHeight))
+{
+  padding = ann::Padding<>(padWLeft, padWRight, padHTop, padHBottom);
+}
+
+template<    
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>&
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::operator=(const Convolution& layer)
+{
+  if (this != &layer)
+  {
+    inSize = layer.inSize; 
+    outSize = layer.outSize; 
+    kernelWidth = layer.kernelWidth; 
+    kernelHeight = layer.kernelHeight; 
+    strideWidth = layer.strideWidth; 
+    strideHeight = layer.strideHeight; 
+    padWLeft = layer.padWLeft; 
+    padWRight = layer.padWRight; 
+    padHBottom = layer.padHBottom; 
+    padHTop = layer.padHTop;
+    weights = layer.weights;  
+    weight = layer.weight;
+    bias = layer.bias; 
+    inputWidth = layer.inputWidth; 
+    inputHeight = layer.inputHeight; 
+    outputWidth = layer.outputWidth;
+    outputHeight = layer.outputHeight;
+    padding = ann::Padding<>(padWLeft, padWRight, padHTop, padHBottom);
+  }
+  return *this; 
+}
+
+template<    
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>& 
+Convolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+> :operator=(Convolution&& layer)
```

Comment:
Hey @gauravghati, this looks like the line that's causing the problem with the build.  If you can get the build worked out, I'd be glad to help move this PR forward and get it merged. :+1: